lib/project/result-view.coffee

```diff
@@ -44,11 +44,61 @@ class ResultView extends View
       @hide()
     else
       @show()
+      @addContextToMatches(@filePath, matches)
       for match in matches
         @matches.append(new MatchView(@model, {@filePath, match}))
 
     @matches.children().eq(selectedIndex).addClass('selected') if selectedIndex > -1
 
+  addContextToMatches: (filePath, matches) ->
+    CONTEXT_LINES = atom.config.get('find-and-replace.searchContextExtraLines')
+    return if CONTEXT_LINES is 0
+    content = fs.readFileSync(filePath).toString()
+    lines = content.split('\n')
```

Comment:
I'm worried about Windows compatibility here - I assume this would still work because CRLF also includes the `\n`, but would the extra `\r` interfere with anything?